cmd/api/src/api/v2/cypherquery.go

```diff
@@ -60,69 +59,69 @@ func (s Resources) CypherSearch(response http.ResponseWriter, request *http.Requ
-			api.WriteErrorResponse(
-				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
-			)
-			return
-		}
+		graphResponse, err = s.cypherMutation(request, preparedQuery, payload.IncludeProperties)
+	} else {
+		graphResponse, err = s.GraphQuery.RawCypherQuery(request.Context(), preparedQuery, payload.IncludeProperties)
+	}
 
-		// create an intent audit log
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
+	if err != nil {
+		switch true {
+		case strings.Contains(err.Error(), "Permission denied: User may not modify the graph."):
 			api.WriteErrorResponse(
 				request.Context(),
-				api.BuildErrorResponse(http.StatusInternalServerError, api.ErrorResponseDetailsInternalServerError, request),
-				response,
+				api.BuildErrorResponse(http.StatusForbidden, err.Error(), request), response,
 			)
-			return
-		}
-	}
-
-	if graphResponse, err := s.GraphQuery.RawCypherSearch(request.Context(), preparedQuery, payload.IncludeProperties); err != nil {
-		// audit log needs to be marked as failed if a mutation is present
-		if preparedQuery.HasMutation {
-			auditLogEntry.Status = model.AuditLogStatusFailure
-		}
-		if queries.IsQueryError(err) {
+		case queries.IsQueryError(err):
 			api.WriteErrorResponse(
 				request.Context(),
 				api.BuildErrorResponse(http.StatusBadRequest, err.Error(), request), response,
 			)
-		} else if util.IsNeoTimeoutError(err) {
+		case util.IsNeoTimeoutError(err):
 			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, "transaction timed out, reduce query complexity or try again later", request), response)
-		} else {
+		default:
 			api.WriteErrorResponse(
 				request.Context(),
 				api.BuildErrorResponse(http.StatusInternalServerError, err.Error(), request), response,
 			)
 		}
 	} else {
-		// audit log needs to be marked as success if a mutation is present
-		if preparedQuery.HasMutation {
-			auditLogEntry.Status = model.AuditLogStatusSuccess
-		}
 		api.WriteBasicResponse(request.Context(), graphResponse, http.StatusOK, response)
 	}
+}
 
-	if preparedQuery.HasMutation {
-		if err := s.DB.AppendAuditLog(request.Context(), auditLogEntry); err != nil {
-			// We don't want to update response as failed because having info on the mutation response trumps this error
-			log.Errorf("failure to create mutation audit log %s", err.Error())
-		}
+func (s Resources) cypherMutation(request *http.Request, preparedQuery queries.PreparedQuery, includeProperties bool) (model.UnifiedGraph, error) {
+	var (
+		auditLogEntry model.AuditEntry
+		graphResponse model.UnifiedGraph
+		err           error
+	)
+
+	if !s.Authorizer.AllowsPermission(ctx.FromRequest(request).AuthCtx, auth.Permissions().GraphDBMutate) {
+		s.Authorizer.AuditLogUnauthorizedAccess(request)
+		return model.UnifiedGraph{}, errors.New("Permission denied: User may not modify the graph.")
```

Comment:
Move this to a global variable to use as a ErrModificationDisallowed (or similar) sentinel